Creating user from config file fails if email has uppercase characters

### Describe the bug

When a user is created from a `config.yml` file, Metabase checks if it already exists in the database by looking at the email address as-is. When inserting the user, the email gets lowercased. This means that importing a user having an email with uppercase characters will always fail.

### To Reproduce

Create a `config.yml` with a user that has uppercase characters in their email address, e.g.:

```yaml
version: 1
config:
  users:
  - first_name: Charly
    last_name: Config
    password: "123123123"
    email: CHARLY@example.com
```

Run Metabase Enterprise and import this user:

```
MB_CONFIG_FILE_PATH=.../config.yml java -jar metabase.jar
```

Note that the user's email gets stored in lowercase:

```
select email from core_user
```

| email |
| :--- |
| internal@metabase.com |
| charly@example.com |

Run Metabase Enterprise with the same config file again. It now fails to start with an error:

<details><summary>Logs</summary>

```
2025-08-28 17:16:16,862 INFO advanced-config.file :: Found config file at path "/Users/rolodato/source/metabase/sandbox/init/config.yml"; Metabase will be initialized with values from this file 🗄️
2025-08-28 17:16:16,865 INFO advanced-config.file :: Initializing :users from config file... 🗄️
2025-08-28 17:16:16,868 INFO file.users :: Creating the first User for this instance. The first user is always created as an admin.
2025-08-28 17:16:16,868 INFO file.users :: Creating new User "Charly Config" with email "CHARLY@example.com"
2025-08-28 17:16:16,948 ERROR core.core :: Metabase Initialization FAILED
clojure.lang.ExceptionInfo: ERROR: duplicate key value violates unique constraint "core_user_email_key"
  Detail: Key (email)=(charly@example.com) already exists. {:toucan2/context-trace [["execute SQL with class com.mchange.v2.c3p0.impl.NewProxyConnection" {:toucan2.jdbc.query/sql-args ["INSERT INTO \"core_user\" (\"email\", \"first_name\", \"password_salt\", \"password\", \"last_login\", \"is_active\", \"updated_at\", \"is_superuser\", \"last_name\", \"date_joined\", \"entity_id\") VALUES (?, ?, ?, ?, NULL, TRUE, NOW(), FALSE, ?, NOW(), ?)" "charly@example.com" "Charly" "4976d027-6b4e-44d0-a3a1-887fccb04f38" "$2a$10$p0Eq9ivX0AecnQEgrxo8zei/hQpHH6A7HtwBZWOTytOOTUyFV.O9W" "Config" "QXgb-gDpzdaE3A24A2sQO"]}] ["resolve connection" {:toucan2.connection/connectable org.postgresql.jdbc.PgConnection}] ["resolve connection" {:toucan2.connection/connectable nil}] {:toucan2.pipeline/rf #object[clojure.core$map$fn__5952$fn__5953 0x1fe7fc2f "clojure.core$map$fn__5952$fn__5953@1fe7fc2f"]} ["with compiled query" {:toucan2.pipeline/compiled-query ["INSERT INTO \"core_user\" (\"email\", \"first_name\", \"password_salt\", \"password\", \"last_login\", \"is_active\", \"updated_at\", \"is_superuser\", \"last_name\", \"date_joined\", \"entity_id\") VALUES (?, ?, ?, ?, NULL, TRUE, NOW(), FALSE, ?, NOW(), ?)" "charly@example.com" "Charly" "4976d027-6b4e-44d0-a3a1-887fccb04f38" "$2a$10$p0Eq9ivX0AecnQEgrxo8zei/hQpHH6A7HtwBZWOTytOOTUyFV.O9W" "Config" "QXgb-gDpzdaE3A24A2sQO"]}] ["with built query" {:toucan2.pipeline/built-query {:insert-into [:core_user], :values ((toucan2.instance/instance :model/User {:email "charly@example.com", :first_name "Charly", :password_salt "4976d027-6b4e-44d0-a3a1-887fccb04f38", :password "$2a$10$p0Eq9ivX0AecnQEgrxo8zei/hQpHH6A7HtwBZWOTytOOTUyFV.O9W", :last_login nil, :is_active true, :updated_at [:metabase.util.honey-sql-2/typed :%now {:database-type "timestamptz"}], :is_superuser false, :last_name "Config", :date_joined :%now, :entity_id "QXgb-gDpzdaE3A24A2sQO"}))}}] ["resolve connection" {:toucan2.connection/connectable org.postgresql.jdbc.PgConnection}] ["resolve connection" {:toucan2.connection/connectable nil}] ["resolve connection" {:toucan2.connection/connectable metabase.app_db.connection.ApplicationDB}] ["resolve connection" {:toucan2.connection/connectable :default}] ["resolve connection" {:toucan2.connection/connectable nil}] ["with resolved query" {:toucan2.pipeline/resolved-query {}}] ["with parsed args" {:toucan2.pipeline/query-type :toucan.query-type/insert.update-count, :toucan2.pipeline/parsed-args {:rows [#ordered/map ([:first_name "Charly"] [:last_name "Config"] [:password "123123123"] [:email "CHARLY@example.com"])]}}] ["with model" {:toucan2.pipeline/model :model/User}] ["with unparsed args" {:toucan2.pipeline/query-type :toucan.query-type/insert.update-count, :toucan2.pipeline/unparsed-args (:model/User #ordered/map ([:first_name "Charly"] [:last_name "Config"] [:password "123123123"] [:email "CHARLY@example.com"]))}]]}
	at org.postgresql.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2734)
	at org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:2421)
	at org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:372)
	at org.postgresql.jdbc.PgStatement.executeInternal(PgStatement.java:518)
	at org.postgresql.jdbc.PgStatement.execute(PgStatement.java:435)
	at org.postgresql.jdbc.PgPreparedStatement.executeWithFlags(PgPreparedStatement.java:196)
	at org.postgresql.jdbc.PgPreparedStatement.execute(PgPreparedStatement.java:182)
	at com.mchange.v2.c3p0.impl.NewProxyPreparedStatement.execute(NewProxyPreparedStatement.java:1216)
	at toucan2.jdbc.query$reduce_jdbc_query.invokeStatic(query.clj:40)
	at toucan2.jdbc.query$reduce_jdbc_query.invoke(query.clj:22)
	at toucan2.jdbc.pipeline$transduce_execute_with_connection_primary_method_java_sql_Connection_default_default.invokeStatic(pipeline.clj:19)
	at toucan2.jdbc.pipeline$transduce_execute_with_connection_primary_method_java_sql_Connection_default_default.invoke(pipeline.clj:9)
	at methodical.impl.combo.common$partial_STAR_$fn__22440.invoke(common.clj:15)
	at methodical.util.FnWithMeta.invoke(util.clj:46)
	at toucan2.jdbc.pipeline$transduce_execute_with_connection_primary_method_java_sql_Connection_toucan_result_type_pks_default.invokeStatic(pipeline.clj:30)
	at toucan2.jdbc.pipeline$transduce_execute_with_connection_primary_method_java_sql_Connection_toucan_result_type_pks_default.invoke(pipeline.clj:22)
	at methodical.impl.combo.common$partial_STAR_$fn__22440.invoke(common.clj:15)
	at methodical.impl.combo.threaded$combine_methods_thread_last$fn__22750$combined_method_thread_last__22751.invoke(threaded.clj:64)
	at methodical.util.FnWithMeta.invoke(util.clj:46)
	at methodical.impl.standard$invoke_multifn.invokeStatic(standard.clj:65)
	at methodical.impl.standard$invoke_multifn.invoke(standard.clj:47)
	at methodical.impl.standard.StandardMultiFn.invoke(standard.clj:216)
	at toucan2.jdbc.pipeline$transduce_execute_with_connection_primary_method_java_sql_Connection_toucan2_jdbc_pipeline_DML_queries_returning_instances_default.invokeStatic(pipeline.clj:62)
	at toucan2.jdbc.pipeline$transduce_execute_with_connection_primary_method_java_sql_Connection_toucan2_jdbc_pipeline_DML_queries_returning_instances_default.invoke(pipeline.clj:52)
	at methodical.impl.combo.common$partial_STAR_$fn__22440.invoke(common.clj:15)
	at methodical.impl.combo.threaded$combine_methods_thread_last$fn__22750$combined_method_thread_last__22751.invoke(threaded.clj:64)
	at methodical.util.FnWithMeta.invoke(util.clj:46)
	at methodical.impl.standard$invoke_multifn.invokeStatic(standard.clj:65)
	at methodical.impl.standard$invoke_multifn.invoke(standard.clj:47)
	at methodical.impl.standard.StandardMultiFn.invoke(standard.clj:216)
	at toucan2.pipeline$transduce_execute$with_connection_STAR___24912$with_transaction_STAR___24913.invoke(pipeline.clj:75)
	at toucan2.connection$bind_current_connectable_fn$fn__24590.invoke(connection.clj:104)
	at metabase.app_db.connection$do_with_transaction_primary_method_java_sql_Connection.invokeStatic(connection.clj:192)
	at metabase.app_db.connection$do_with_transaction_primary_method_java_sql_Connection.invoke(connection.clj:174)
	at methodical.impl.combo.common$partial_STAR_$fn__22440.invoke(common.clj:13)
	at methodical.util.FnWithMeta.invoke(util.clj:46)
	at toucan2.connection$do_with_transaction_around_method_toucan2_connection_default.invokeStatic(connection.clj:249)
	at toucan2.connection$do_with_transaction_around_method_toucan2_connection_default.invoke(connection.clj:245)
	at methodical.impl.combo.common$partial_STAR_$fn__22440.invoke(common.clj:13)
	at methodical.util.FnWithMeta.invoke(util.clj:46)
	at methodical.impl.standard$invoke_multifn.invokeStatic(standard.clj:58)
	at methodical.impl.standard$invoke_multifn.invoke(standard.clj:47)
	at methodical.impl.standard.StandardMultiFn.invoke(standard.clj:212)
	at toucan2.pipeline$transduce_execute$with_connection_STAR___24912.invoke(pipeline.clj:74)
	at toucan2.connection$bind_current_connectable_fn$fn__24590.invoke(connection.clj:104)
	at toucan2.connection$bind_current_connectable_fn$fn__24590.invoke(connection.clj:104)
	at toucan2.jdbc.connection$do_with_connection_primary_method_java_sql_Connection.invokeStatic(connection.clj:13)
	at toucan2.jdbc.connection$do_with_connection_primary_method_java_sql_Connection.invoke(connection.clj:11)
	at methodical.impl.combo.common$partial_STAR_$fn__22440.invoke(common.clj:12)
	at methodical.util.FnWithMeta.invoke(util.clj:46)
	at toucan2.connection$do_with_connection_around_method_toucan2_connection_default.invokeStatic(connection.clj:118)
	at toucan2.connection$do_with_connection_around_method_toucan2_connection_default.invoke(connection.clj:106)
	at methodical.impl.combo.common$partial_STAR_$fn__22440.invoke(common.clj:12)
	at methodical.util.FnWithMeta.invoke(util.clj:46)
	at methodical.impl.standard$invoke_multifn.invokeStatic(standard.clj:55)
	at methodical.impl.standard$invoke_multifn.invoke(standard.clj:47)
	at methodical.impl.standard.StandardMultiFn.invoke(standard.clj:210)
	at toucan2.connection$do_with_connection_primary_method_.invokeStatic(connection.clj:204)
	at toucan2.connection$do_with_connection_primary_method_.invoke(connection.clj:194)
	at methodical.impl.combo.common$partial_STAR_$fn__22440.invoke(common.clj:12)
	at methodical.util.FnWithMeta.invoke(util.clj:46)
	at toucan2.connection$do_with_connection_around_method_toucan2_connection_default.invokeStatic(connection.clj:118)
	at toucan2.connection$do_with_connection_around_method_toucan2_connection_default.invoke(connection.clj:106)
	at methodical.impl.combo.common$partial_STAR_$fn__22440.invoke(common.clj:12)
	at methodical.util.FnWithMeta.invoke(util.clj:46)
	at methodical.impl.standard$invoke_multifn.invokeStatic(standard.clj:55)
	at methodical.impl.standard$invoke_multifn.invoke(standard.clj:47)
	at methodical.impl.standard.StandardMultiFn.invoke(standard.clj:210)
	at toucan2.pipeline$transduce_execute.invokeStatic(pipeline.clj:74)
	at toucan2.pipeline$transduce_execute.invoke(pipeline.clj:64)
	at clojure.lang.Var.invoke(Var.java:401)
	at toucan2.pipeline$transduce_compiled_query.invokeStatic(pipeline.clj:244)
	at toucan2.pipeline$transduce_compiled_query.invoke(pipeline.clj:240)
	at toucan2.pipeline$transduce_built_query.invokeStatic(pipeline.clj:252)
	at toucan2.pipeline$transduce_built_query.invoke(pipeline.clj:246)
	at toucan2.pipeline$transduce_query_primary_method_default.invokeStatic(pipeline.clj:272)
	at toucan2.pipeline$transduce_query_primary_method_default.invoke(pipeline.clj:269)
	at methodical.impl.combo.common$partial_STAR_$fn__22440.invoke(common.clj:15)
	at methodical.util.FnWithMeta.invoke(util.clj:46)
	at toucan2.tools.after$transduce_query_primary_method_toucan2_tools_after_query_type_toucan2_tools_after_model_default.invokeStatic(after.clj:95)
	at toucan2.tools.after$transduce_query_primary_method_toucan2_tools_after_query_type_toucan2_tools_after_model_default.invoke(after.clj:84)
	at methodical.impl.combo.common$partial_STAR_$fn__22440.invoke(common.clj:15)
	at methodical.util.FnWithMeta.invoke(util.clj:46)
	at toucan2.tools.default_fields$transduce_query_primary_method_toucan_result_type_instances_toucan2_tools_default_fields_default_fields_clojure_lang_IPersistentMap.invokeStatic(default_fields.clj:67)
	at toucan2.tools.default_fields$transduce_query_primary_method_toucan_result_type_instances_toucan2_tools_default_fields_default_fields_clojure_lang_IPersistentMap.invoke(default_fields.clj:61)
	at methodical.impl.combo.common$partial_STAR_$fn__22440.invoke(common.clj:15)
	at methodical.impl.combo.threaded$combine_methods_thread_last$fn__22750$combined_method_thread_last__22751.invoke(threaded.clj:64)
	at methodical.util.FnWithMeta.invoke(util.clj:46)
	at toucan2.tools.before_insert$transduce_query_around_method_toucan_query_type_insert__STAR__toucan2_tools_before_insert_before_insert_default$with_connection_STAR___29116$with_transaction_STAR___29117.invoke(before_insert.clj:37)
	at toucan2.connection$bind_current_connectable_fn$fn__24590.invoke(connection.clj:104)
	at metabase.app_db.connection$do_with_transaction_primary_method_java_sql_Connection.invokeStatic(connection.clj:192)
	at metabase.app_db.connection$do_with_transaction_primary_method_java_sql_Connection.invoke(connection.clj:174)
	at methodical.impl.combo.common$partial_STAR_$fn__22440.invoke(common.clj:13)
	at methodical.util.FnWithMeta.invoke(util.clj:46)
	at toucan2.connection$do_with_transaction_around_method_toucan2_connection_default.invokeStatic(connection.clj:249)
	at toucan2.connection$do_with_transaction_around_method_toucan2_connection_default.invoke(connection.clj:245)
	at methodical.impl.combo.common$partial_STAR_$fn__22440.invoke(common.clj:13)
	at methodical.util.FnWithMeta.invoke(util.clj:46)
	at methodical.impl.standard$invoke_multifn.invokeStatic(standard.clj:58)
	at methodical.impl.standard$invoke_multifn.invoke(standard.clj:47)
	at methodical.impl.standard.StandardMultiFn.invoke(standard.clj:212)
	at toucan2.tools.before_insert$transduce_query_around_method_toucan_query_type_insert__STAR__toucan2_tools_before_insert_before_insert_default$with_connection_STAR___29116.invoke(before_insert.clj:36)
	at toucan2.connection$bind_current_connectable_fn$fn__24590.invoke(connection.clj:104)
	at toucan2.connection$bind_current_connectable_fn$fn__24590.invoke(connection.clj:104)
	at toucan2.jdbc.connection$do_with_connection_primary_method_java_sql_Connection.invokeStatic(connection.clj:13)
	at toucan2.jdbc.connection$do_with_connection_primary_method_java_sql_Connection.invoke(connection.clj:11)
	at methodical.impl.combo.common$partial_STAR_$fn__22440.invoke(common.clj:12)
	at methodical.util.FnWithMeta.invoke(util.clj:46)
	at toucan2.connection$do_with_connection_around_method_toucan2_connection_default.invokeStatic(connection.clj:118)
	at toucan2.connection$do_with_connection_around_method_toucan2_connection_default.invoke(connection.clj:106)
	at methodical.impl.combo.common$partial_STAR_$fn__22440.invoke(common.clj:12)
	at methodical.util.FnWithMeta.invoke(util.clj:46)
	at methodical.impl.standard$invoke_multifn.invokeStatic(standard.clj:55)
	at methodical.impl.standard$invoke_multifn.invoke(standard.clj:47)
	at methodical.impl.standard.StandardMultiFn.invoke(standard.clj:210)
	at toucan2.connection$do_with_connection_primary_method_.invokeStatic(connection.clj:204)
	at toucan2.connection$do_with_connection_primary_method_.invoke(connection.clj:194)
	at methodical.impl.combo.common$partial_STAR_$fn__22440.invoke(common.clj:12)
	at methodical.util.FnWithMeta.invoke(util.clj:46)
	at toucan2.connection$do_with_connection_around_method_toucan2_connection_default.invokeStatic(connection.clj:118)
	at toucan2.connection$do_with_connection_around_method_toucan2_connection_default.invoke(connection.clj:106)
	at methodical.impl.combo.common$partial_STAR_$fn__22440.invoke(common.clj:12)
	at methodical.util.FnWithMeta.invoke(util.clj:46)
	at methodical.impl.standard$invoke_multifn.invokeStatic(standard.clj:55)
	at methodical.impl.standard$invoke_multifn.invoke(standard.clj:47)
	at methodical.impl.standard.StandardMultiFn.invoke(standard.clj:210)
	at toucan2.tools.before_insert$transduce_query_around_method_toucan_query_type_insert__STAR__toucan2_tools_before_insert_before_insert_default.invokeStatic(before_insert.clj:36)
	at toucan2.tools.before_insert$transduce_query_around_method_toucan_query_type_insert__STAR__toucan2_tools_before_insert_before_insert_default.invoke(before_insert.clj:31)
	at methodical.impl.combo.common$partial_STAR_$fn__22440.invoke(common.clj:15)
	at methodical.util.FnWithMeta.invoke(util.clj:46)
	at methodical.impl.standard$invoke_multifn.invokeStatic(standard.clj:65)
	at methodical.impl.standard$invoke_multifn.invoke(standard.clj:47)
	at methodical.impl.standard.StandardMultiFn.invoke(standard.clj:216)
	at toucan2.tools.after$transduce_query_primary_method_toucan2_tools_after_query_type_toucan2_tools_after_model_default.invokeStatic(after.clj:102)
	at toucan2.tools.after$transduce_query_primary_method_toucan2_tools_after_query_type_toucan2_tools_after_model_default.invoke(after.clj:84)
	at methodical.impl.combo.common$partial_STAR_$fn__22440.invoke(common.clj:15)
	at methodical.impl.combo.threaded$combine_methods_thread_last$fn__22750$combined_method_thread_last__22751.invoke(threaded.clj:64)
	at methodical.util.FnWithMeta.invoke(util.clj:46)
	at toucan2.tools.before_insert$transduce_query_around_method_toucan_query_type_insert__STAR__toucan2_tools_before_insert_before_insert_default$with_connection_STAR___29116$with_transaction_STAR___29117.invoke(before_insert.clj:37)
	at toucan2.connection$bind_current_connectable_fn$fn__24590.invoke(connection.clj:104)
	at metabase.app_db.connection$do_transaction$thunk__39530.invoke(connection.clj:145)
	at metabase.app_db.connection$do_transaction.invokeStatic(connection.clj:164)
	at metabase.app_db.connection$do_transaction.invoke(connection.clj:141)
	at metabase.app_db.connection$do_with_transaction_primary_method_java_sql_Connection.invokeStatic(connection.clj:201)
	at metabase.app_db.connection$do_with_transaction_primary_method_java_sql_Connection.invoke(connection.clj:174)
	at methodical.impl.combo.common$partial_STAR_$fn__22440.invoke(common.clj:13)
	at methodical.util.FnWithMeta.invoke(util.clj:46)
	at toucan2.connection$do_with_transaction_around_method_toucan2_connection_default.invokeStatic(connection.clj:249)
	at toucan2.connection$do_with_transaction_around_method_toucan2_connection_default.invoke(connection.clj:245)
	at methodical.impl.combo.common$partial_STAR_$fn__22440.invoke(common.clj:13)
	at methodical.util.FnWithMeta.invoke(util.clj:46)
	at methodical.impl.standard$invoke_multifn.invokeStatic(standard.clj:58)
	at methodical.impl.standard$invoke_multifn.invoke(standard.clj:47)
	at methodical.impl.standard.StandardMultiFn.invoke(standard.clj:212)
	at toucan2.tools.before_insert$transduce_query_around_method_toucan_query_type_insert__STAR__toucan2_tools_before_insert_before_insert_default$with_connection_STAR___29116.invoke(before_insert.clj:36)
	at toucan2.connection$bind_current_connectable_fn$fn__24590.invoke(connection.clj:104)
	at toucan2.connection$bind_current_connectable_fn$fn__24590.invoke(connection.clj:104)
	at toucan2.connection$bind_current_connectable_fn$fn__24590.invoke(connection.clj:104)
	at toucan2.jdbc.connection$do_with_connection_primary_method_javax_sql_DataSource.invokeStatic(connection.clj:18)
	at toucan2.jdbc.connection$do_with_connection_primary_method_javax_sql_DataSource.invoke(connection.clj:15)
	at methodical.impl.combo.common$partial_STAR_$fn__22440.invoke(common.clj:12)
	at methodical.util.FnWithMeta.invoke(util.clj:46)
	at toucan2.connection$do_with_connection_around_method_toucan2_connection_default.invokeStatic(connection.clj:118)
	at toucan2.connection$do_with_connection_around_method_toucan2_connection_default.invoke(connection.clj:106)
	at methodical.impl.combo.common$partial_STAR_$fn__22440.invoke(common.clj:12)
	at methodical.util.FnWithMeta.invoke(util.clj:46)
	at methodical.impl.standard$invoke_multifn.invokeStatic(standard.clj:55)
	at methodical.impl.standard$invoke_multifn.invoke(standard.clj:47)
	at methodical.impl.standard.StandardMultiFn.invoke(standard.clj:210)
	at metabase.app_db.connection$do_with_connection_primary_method_default.invokeStatic(connection.clj:132)
	at metabase.app_db.connection$do_with_connection_primary_method_default.invoke(connection.clj:130)
	at methodical.impl.combo.common$partial_STAR_$fn__22440.invoke(common.clj:12)
	at methodical.util.FnWithMeta.invoke(util.clj:46)
	at toucan2.connection$do_with_connection_around_method_toucan2_connection_default.invokeStatic(connection.clj:118)
	at toucan2.connection$do_with_connection_around_method_toucan2_connection_default.invoke(connection.clj:106)
	at methodical.impl.combo.common$partial_STAR_$fn__22440.invoke(common.clj:12)
	at methodical.util.FnWithMeta.invoke(util.clj:46)
	at methodical.impl.standard$invoke_multifn.invokeStatic(standard.clj:55)
	at methodical.impl.standard$invoke_multifn.invoke(standard.clj:47)
	at methodical.impl.standard.StandardMultiFn.invoke(standard.clj:210)
	at toucan2.connection$do_with_connection_primary_method_.invokeStatic(connection.clj:204)
	at toucan2.connection$do_with_connection_primary_method_.invoke(connection.clj:194)
	at methodical.impl.combo.common$partial_STAR_$fn__22440.invoke(common.clj:12)
	at methodical.util.FnWithMeta.invoke(util.clj:46)
	at toucan2.connection$do_with_connection_around_method_toucan2_connection_default.invokeStatic(connection.clj:118)
	at toucan2.connection$do_with_connection_around_method_toucan2_connection_default.invoke(connection.clj:106)
	at methodical.impl.combo.common$partial_STAR_$fn__22440.invoke(common.clj:12)
	at methodical.util.FnWithMeta.invoke(util.clj:46)
	at methodical.impl.standard$invoke_multifn.invokeStatic(standard.clj:55)
	at methodical.impl.standard$invoke_multifn.invoke(standard.clj:47)
	at methodical.impl.standard.StandardMultiFn.invoke(standard.clj:210)
	at toucan2.tools.before_insert$transduce_query_around_method_toucan_query_type_insert__STAR__toucan2_tools_before_insert_before_insert_default.invokeStatic(before_insert.clj:36)
	at toucan2.tools.before_insert$transduce_query_around_method_toucan_query_type_insert__STAR__toucan2_tools_before_insert_before_insert_default.invoke(before_insert.clj:31)
	at methodical.impl.combo.common$partial_STAR_$fn__22440.invoke(common.clj:15)
	at methodical.util.FnWithMeta.invoke(util.clj:46)
	at methodical.impl.standard$invoke_multifn.invokeStatic(standard.clj:65)
	at methodical.impl.standard$invoke_multifn.invoke(standard.clj:47)
	at methodical.impl.standard.StandardMultiFn.invoke(standard.clj:216)
	at toucan2.pipeline$transduce_query_STAR_.invokeStatic(pipeline.clj:278)
	at toucan2.pipeline$transduce_query_STAR_.invoke(pipeline.clj:274)
	at toucan2.pipeline$transduce_with_model.invokeStatic(pipeline.clj:293)
	at toucan2.pipeline$transduce_with_model.invoke(pipeline.clj:280)
	at toucan2.pipeline$transduce_parsed.invokeStatic(pipeline.clj:309)
	at toucan2.pipeline$transduce_parsed.invoke(pipeline.clj:295)
	at toucan2.pipeline$transduce_unparsed.invokeStatic(pipeline.clj:317)
	at toucan2.pipeline$transduce_unparsed.invoke(pipeline.clj:311)
	at toucan2.pipeline$transduce_unparsed_with_default_rf.invokeStatic(pipeline.clj:392)
	at toucan2.pipeline$transduce_unparsed_with_default_rf.invoke(pipeline.clj:386)
	at toucan2.insert$insert_BANG_.invokeStatic(insert.clj:124)
	at toucan2.insert$insert_BANG_.doInvoke(insert.clj:74)
	at clojure.lang.RestFn.invoke(RestFn.java:424)
	at metabase_enterprise.advanced_config.file.users$init_from_config_file_BANG_.invokeStatic(users.clj:54)
	at metabase_enterprise.advanced_config.file.users$init_from_config_file_BANG_.invoke(users.clj:38)
	at metabase_enterprise.advanced_config.file.users$fn__133085.invokeStatic(users.clj:59)
	at metabase_enterprise.advanced_config.file.users$fn__133085.invoke(users.clj:56)
	at clojure.lang.MultiFn.invoke(MultiFn.java:234)
	at metabase_enterprise.advanced_config.file$initialize_BANG_.invokeStatic(file.clj:268)
	at metabase_enterprise.advanced_config.file$initialize_BANG_.invoke(file.clj:253)
	at clojure.lang.Var.invoke(Var.java:382)
	at metabase.core.config_from_file$init_from_file_if_code_available_BANG_.invokeStatic(config_from_file.clj:17)
	at metabase.core.config_from_file$init_from_file_if_code_available_BANG_.invoke(config_from_file.clj:6)
	at metabase.core.core$init_BANG__STAR_.invokeStatic(core.clj:156)
	at metabase.core.core$init_BANG__STAR_.invoke(core.clj:117)
	at metabase.core.core$init_BANG_.invokeStatic(core.clj:194)
	at metabase.core.core$init_BANG_.invoke(core.clj:189)
	at metabase.core.core$start_normally.invokeStatic(core.clj:208)
	at metabase.core.core$start_normally.invoke(core.clj:200)
	at metabase.core.core$entrypoint.invokeStatic(core.clj:243)
	at metabase.core.core$entrypoint.doInvoke(core.clj:234)
	at clojure.lang.RestFn.invoke(RestFn.java:400)
	at clojure.lang.AFn.applyToHelper(AFn.java:152)
	at clojure.lang.RestFn.applyTo(RestFn.java:135)
	at clojure.lang.Var.applyTo(Var.java:707)
	at clojure.core$apply.invokeStatic(core.clj:667)
	at clojure.core$apply.invoke(core.clj:662)
	at metabase.core.bootstrap$_main.invokeStatic(bootstrap.clj:36)
	at metabase.core.bootstrap$_main.doInvoke(bootstrap.clj:29)
	at clojure.lang.RestFn.invoke(RestFn.java:400)
	at clojure.lang.AFn.applyToHelper(AFn.java:152)
	at clojure.lang.RestFn.applyTo(RestFn.java:135)
	at metabase.core.bootstrap.main(Unknown Source)
Caused by: org.postgresql.util.PSQLException: ERROR: duplicate key value violates unique constraint "core_user_email_key"
  Detail: Key (email)=(charly@example.com) already exists.
	... 241 more
2025-08-28 17:16:16,955 INFO core.core :: Metabase Shutting Down ...
```

</details> 

### Expected behavior

Either convert an email to lowercase before checking if it already exists, or store emails from config files as-is without lowercasing. Don't cause errors.

### Logs

_No response_

### Information about your Metabase installation

```JSON
{
  "browser-info": {
    "language": "en-US",
    "platform": "MacIntel",
    "userAgent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:141.0) Gecko/20100101 Firefox/141.0",
    "vendor": ""
  },
  "metabase-info": {
    "databases": [
      "postgres",
      "h2"
    ],
    "run-mode": "dev",
    "plan-alias": "pro-self-hosted",
    "version": {
      "date": "2025-08-22",
      "tag": "v1.56.3",
      "hash": "026cbed"
    },
    "settings": {
      "report-timezone": null
    },
    "hosting-env": "unknown",
    "application-database": "postgres",
    "application-database-details": {
      "database": {
        "name": "PostgreSQL",
        "version": "17.5 (Debian 17.5-1.pgdg120+1)"
      },
      "jdbc-driver": {
        "name": "PostgreSQL JDBC Driver",
        "version": "42.7.7"
      }
    }
  },
  "system-info": {
    "file.encoding": "UTF-8",
    "java.runtime.name": "OpenJDK Runtime Environment",
    "java.runtime.version": "21.0.7+6-LTS",
    "java.vendor": "Eclipse Adoptium",
    "java.vendor.url": "https://adoptium.net/",
    "java.version": "21.0.7",
    "java.vm.name": "OpenJDK 64-Bit Server VM",
    "java.vm.version": "21.0.7+6-LTS",
    "os.name": "Mac OS X",
    "os.version": "15.1",
    "user.language": "en",
    "user.timezone": "America/Argentina/Buenos_Aires"
  }
}
```

### Severity

Annoying, but has a workaround (make the email all lowercase in the config file)

### Additional context

_No response_

--- a/enterprise/backend/src/metabase_enterprise/advanced_config/file/users.clj
+++ b/enterprise/backend/src/metabase_enterprise/advanced_config/file/users.clj
@@ -56,4 +56,5 @@
 (defmethod advanced-config.file.i/initialize-section! :users
   [_section-name users]
   (doseq [user users]
-    (init-from-config-file! user)))
+    ;; we're lower-casing emails in :model/User, so we should do the same here
+    (init-from-config-file! (update user :email u/lower-case-en))))

--- a/enterprise/backend/test/metabase_enterprise/advanced_config/file/users_test.clj
+++ b/enterprise/backend/test/metabase_enterprise/advanced_config/file/users_test.clj
@@ -157,3 +157,21 @@
      :last_name  "Era"
      :email      "cam+config-file-test@metabase.com"}
     (re-pattern (java.util.regex.Pattern/quote "failed: (contains? % :password)"))))
+
+(deftest email-case-irrelevant-test
+  (testing "Using any case in email still works #62894"
+    (try
+      (binding [advanced-config.file/*config* {:version 1
+                                               :config  {:users [{:first_name       "Cam"
+                                                                  :last_name        "Era"
+                                                                  :email            "Cam+config-file-test@metabase.com"
+                                                                  :password         "2cans"
+                                                                  :login_attributes {"a" 1}}]}}]
+        (is (= :ok
+               (advanced-config.file/initialize!)))
+        (testing "It should not fail with 'duplicate'"
+          (is (= :ok
+                 (advanced-config.file/initialize!)))))
+      (finally
+        (t2/delete! :model/User :email "cam+config-file-test@metabase.com")
+        (t2/delete! :model/User :email "Cam+config-file-test@metabase.com")))))

